src/main/java/com/socialmedia/dto/post/PostDtoIn.java

```diff
@@ -2,16 +2,14 @@
 
 import com.socialmedia.dto.comment.CommentDtoIn;
 import com.socialmedia.dto.user.UserLabelDtoIn;
+import lombok.Data;
 
 import java.util.List;
 
+@Data
 public class PostDtoIn {
-  private Long id;
   private String message;
-  private Long date;
   private String image;
   private Boolean showEveryone;
-  private UserLabelDtoIn author;
-  private List<CommentDtoIn> comments;
-  private List<UserLabelDtoIn> likes;
+  private String authorId;
```

Comment:
Yep, we do need it.
In entityOf () we set author like this:
ApplicationUser author = userMapper.entityOf(dtoIn.getAuthorId());

Reply:
If it bad practice, could we get Principal in a mapper?

Reply:
As username is id for ApplicationUser entity, you may not include authorId in request dto because it is equal to principal name and you may get it from static SecurityContextHolder everywhere in app. So you may remove authorId field from PostDtoIn and replace in PostMapper this line
ApplicationUser author = userMapper.entityOf(dtoIn.getAuthorId());
with this one
ApplicationUser author = userMapper.entityOf(SecurityContextHolder.getContext().getAuthentication().getName());
I guess you even should do this, because if you get author from request dto, I am able to get auth token for my profile and send via request via postman specifying other user id

Reply:
Thanks!